src/libcsg/modules/io/dlpolytopologyreader.cc

```diff
@@ -219,7 +219,8 @@ bool DLPOLYTopologyReader::ReadTopology(string file, Topology &top) {
 #endif
 
       // read molecule
-      int id_map[natoms];
+      unique_ptr<long int[]> id_map =
+          unique_ptr<long int[]>{new long int[natoms]};
```

Comment:
@JensWehner please check.

Reply:
that sounds very complicated what we want is a std::vector<int>(natoms);